File: sample/src/main/java/me/xiaopan/assemblyadaptersample/itemfactory/HeaderItemFactory.java
package me.xiaopan.assemblyadaptersample.itemfactory;

import android.content.Context;
import android.view.ViewGroup;
import android.widget.TextView;

import me.xiaopan.assemblyadapter.AssemblyItem;
import me.xiaopan.assemblyadapter.AssemblyItemFactory;
import me.xiaopan.assemblyadaptersample.R;

public class HeaderItemFactory extends AssemblyItemFactory<HeaderItemFactory.HeaderItem>{
    @Override
    public boolean isTarget(Object itemObject) {
        return itemObject instanceof String;
    }

    @Override
    public HeaderItem createAssemblyItem(ViewGroup parent) {
        return new HeaderItem(R.layout.list_item_header, parent);
    }

    public class HeaderItem extends AssemblyItem<String>{
        private TextView textView;

        public HeaderItem(int itemLayoutId, ViewGroup parent) {
            super(itemLayoutId, parent);
        }

        @Override
        protected void onFindViews() {
            textView = (TextView) getItemView();
        }

        @Override
        protected void onConfigViews(Context context) {

        }

        @Override
        protected void onSetData(int position, String s) {
            textView.setText(s);
        }
    }
}

File: sample/src/main/java/me/xiaopan/assemblyadaptersample/itemfactory/HeaderItemFactory.kt
package me.xiaopan.assemblyadaptersample.itemfactory

import android.content.Context
import android.view.ViewGroup
import android.widget.TextView
import me.xiaopan.assemblyadapter.AssemblyItem
import me.xiaopan.assemblyadapter.AssemblyItemFactory
import me.xiaopan.assemblyadaptersample.R

class HeaderItemFactory : AssemblyItemFactory<HeaderItemFactory.HeaderItem>() {
    override fun isTarget(data: Any): Boolean {
        return data is String
    }

    override fun createAssemblyItem(parent: ViewGroup): HeaderItem {
        return HeaderItem(R.layout.list_item_header, parent)
    }

    inner class HeaderItem(itemLayoutId: Int, parent: ViewGroup) : AssemblyItem<String>(itemLayoutId, parent) {

        override fun onConfigViews(context: Context) {

        }

        override fun onSetData(position: Int, s: String) {
            (itemView as TextView).text = s
        }
    }
}
